src/parser/deserializer.ts

```diff
@@ -45,6 +45,8 @@ export interface DeserializeOptions {
   index?: number;
 
   raw?: boolean;
+  /** allows for opt-in utf-8 validation */
```

Comment:
Let's expand this documentation a bit more. First off, IIRC this allows for opt-out, since the default is to always validate. Also, I think the type should be `utf8: boolean | Record<string, boolean>` right? cus it can just take an overall setting? Additionally, we could use some examples of how to use this option correctly.

(You can use the `@example` tsdoc block for this!)